src/plot.cpp

```diff
@@ -637,51 +639,80 @@ Plot::Plot(pugi::xml_node plot_node)
-// POSITION_RGB computes the red/green/blue values for a given plot with the
-// current particle's position
-//==============================================================================
+template<class D>
+D PlotBase::generate_data() const {
 
+  size_t width = pixels_[0];
+  size_t height = pixels_[1];
 
-void position_rgb(Particle p, Plot pl, RGBColor& rgb, int& id)
-{
-  p.n_coord_ = 1;
+  // get pixel size
+  double in_pixel = (width_[0])/static_cast<double>(width);
+  double out_pixel = (width_[1])/static_cast<double>(height);
 
-  bool found_cell = find_cell(&p, 0);
+  // size data array
+  D data(width, height);
 
-  int j = p.n_coord_ - 1;
+  // setup basis indices and initial position centered on pixel
+  int in_i, out_i;
+  Position xyz = origin_;
+  switch(basis_) {
+  case PlotBasis::xy :
+    in_i = 0;
+    out_i = 1;
+    break;
+  case PlotBasis::xz :
+    in_i = 0;
+    out_i = 2;
+    break;
+  case PlotBasis::yz :
+    in_i = 1;
+    out_i = 2;
+    break;
+  }
 
-  if (settings::check_overlaps) {check_cell_overlap(&p);}
+  // set initial position
+  xyz[in_i] = origin_[in_i] - width_[0] / 2. + in_pixel / 2.;
+  xyz[out_i] = origin_[out_i] + width_[1] / 2. - out_pixel / 2.;
 
-  // Set coordinate level if specified
-  if (pl.level_ >= 0) {j = pl.level_ + 1;}
+  // arbitrary direction
+  Direction dir = {0.5, 0.5, 0.5};
 
-  if (!found_cell) {
-    // If no cell, revert to default color
-    rgb = pl.not_found_;
-    id = NOT_FOUND;
-  } else {
-    if (PlotColorBy::mats == pl.color_by_) {
-      // Assign color based on material
-      const auto& c = model::cells[p.coord_[j].cell];
-      if (c->type_ == FILL_UNIVERSE) {
-        // If we stopped on a middle universe level, treat as if not found
-        rgb = pl.not_found_;
-        id = NOT_FOUND;
-      } else if (p.material_ == MATERIAL_VOID) {
-        // By default, color void cells white
-        rgb = WHITE;
-        id = NOT_FOUND;
-      } else {
-        rgb = pl.colors_[p.material_];
-        id = model::materials[p.material_]->id_;
+#pragma omp parallel
+{
+  Particle p;
+  p.r() = xyz;
+  p.u() = dir;
+  p.coord_[0].universe = model::root_universe;
+  int level = level_;
+  int j{};
+
+  #pragma omp for
+  for (int y = 0; y < height; y++) {
+    p.r()[out_i] =  xyz[out_i] - out_pixel * y;
+    for (int x = 0; x < width; x++) {
+      p.r()[in_i] = xyz[in_i] + in_pixel * x;
+      p.n_coord_ = 1;
+      // local variables
+      bool found_cell = find_cell(&p, 0);
+      j = p.n_coord_ - 1;
+      if (level >=0) {j = level + 1;}
+      if (found_cell) {
+        data.set_value(y, x, p, j);
+        Cell* c = model::cells[p.coord_[j].cell].get();
       }
-    } else if (PlotColorBy::cells == pl.color_by_) {
-      // Assign color based on cell
-      rgb = pl.colors_[p.coord_[j].cell];
-      id = model::cells[p.coord_[j].cell]->id_;
-    }
-  } // endif found_cell
+    } // inner for
+  } // outer for
+ } // omp parallel
+
+ return data;
```

Comment:
When you fix the pragma indentation, be mindful of indentation down here too (this is only one space out)